src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -410,15 +413,16 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             throws IOException, InterruptedException {
         setEnv(build.getEnvironment(listener));
         final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
+        FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
```

Comment:
moves closer to the code where we launch and execute

Reply:
Refactored to read in procedural way that is suggested